File: maui-sample/Models/LocationV2DataMessage.cs
namespace MauiSample.Models;

public sealed class LocationV2DataMessage
{
  public double? Latitude { get; init; }

  public double? Longitude { get; init; }

  public double? Altitude { get; init; }

  public double? Speed { get; init; }

  public double? Bearing { get; init; }

  public string? SolutionType { get; init; }

  public double? Hrms { get; init; }

  public double? Vrms { get; init; }
}

File: android-native-sample/app/src/main/java/com/example/kotlin_sample/models/LocationV2DataMessage.kt
package com.example.kotlin_sample.models

import kotlinx.serialization.Serializable

@Serializable
data class LocationV2DataMessage(
  val latitude: Double? = null,
  val longitude: Double? = null,
  val altitude: Double? = null,
  val speed: Double? = null,
  val bearing: Double? = null,
  val solutionType: String? = null,
  val hrms: Double? = null,
  val vrms: Double? = null,
)
